obstore/python/obstore/fsspec.py

```diff
@@ -192,31 +324,62 @@ async def _put_file(
         **_kwargs: Any,
     ) -> None:
+        if not Path(lpath).is_file():
+            err_msg = f"File {lpath} not found in local"
+            raise FileNotFoundError(err_msg)
+
         # TODO: convert to use async file system methods using LocalStore
         # Async functions should not open files with blocking methods like `open`
+        rbucket, rpath = self._split_path(rpath)
+
+        # Should construct the store instance by rbucket, which is the target path
+        store = self._construct_store(rbucket)
+
         with open(lpath, "rb") as f:  # noqa: ASYNC230
-            await obs.put_async(self.store, rpath, f)
+            await obs.put_async(store, rpath, f)
 
     async def _get_file(self, rpath: str, lpath: str, **_kwargs: Any) -> None:
+        res = urlparse(lpath)
+        if res.scheme or Path(lpath).is_dir():
+            # lpath need to be local file and cannot contain scheme
+            return
+
         # TODO: convert to use async file system methods using LocalStore
         # Async functions should not open files with blocking methods like `open`
+        rbucket, rpath = self._split_path(rpath)
+
+        # Should construct the store instance by rbucket, which is the target path
+        store = self._construct_store(rbucket)
+
         with open(lpath, "wb") as f:  # noqa: ASYNC230
-            resp = await obs.get_async(self.store, rpath)
+            resp = await obs.get_async(store, rpath)
             async for buffer in resp.stream():
                 f.write(buffer)
 
     async def _info(self, path: str, **_kwargs: Any) -> dict[str, Any]:
-        head = await obs.head_async(self.store, path)
-        return {
-            # Required of `info`: (?)
-            "name": head["path"],
-            "size": head["size"],
-            "type": "directory" if head["path"].endswith("/") else "file",
-            # Implementation-specific keys
-            "e_tag": head["e_tag"],
-            "last_modified": head["last_modified"],
-            "version": head["version"],
-        }
+        bucket, path_no_bucket = self._split_path(path)
+        store = self._construct_store(bucket)
+
+        try:
+            head = await obs.head_async(store, path_no_bucket)
+            return {
+                # Required of `info`: (?)
+                "name": head["path"],
+                "size": head["size"],
+                "type": "directory" if head["path"].endswith("/") else "file",
+                # Implementation-specific keys
+                "e_tag": head["e_tag"],
+                "last_modified": head["last_modified"],
+                "version": head["version"],
+            }
+        except FileNotFoundError:
+            # use info in fsspec.AbstractFileSystem
+            loop = asyncio.get_running_loop()
+            return await loop.run_in_executor(None, super().info, path, **_kwargs)
+
+    @staticmethod
+    def _fill_bucket_name(path: str, bucket: str) -> str:
```

Comment:
This is only used in two places. Can we just copy the f-string above and delete this helper function?